src/test/java/com/example/demo/helpers/HoverflyService.java

```diff
@@ -1,18 +1,45 @@
 package com.example.demo.helpers;
 
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
 import io.specto.hoverfly.junit.core.Hoverfly;
 import io.specto.hoverfly.junit.core.HoverflyConfig;
 import io.specto.hoverfly.junit.core.HoverflyMode;
+import javax.net.ssl.SSLException;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.transport.ProxyProvider.Proxy;
 
 @Configuration
 public class HoverflyService {
 
     @Bean
     public Hoverfly hoverfly() {
-        Hoverfly hv = new Hoverfly(HoverflyConfig.localConfigs().proxyLocalHost(), HoverflyMode.SIMULATE);
+        Hoverfly hv = new Hoverfly(HoverflyConfig.localConfigs(), HoverflyMode.SIMULATE);
```

Comment:
if you don't use `proxyLocalHost` it won't capture `localhost` traffic, that's why I've deleted the redundant `BypassHoverflyProxySelector`